include/constants/heap.h

```diff
@@ -1,6 +1,14 @@
 #ifndef POKEPLATINUM_CONSTANTS_HEAP_H
 #define POKEPLATINUM_CONSTANTS_HEAP_H
 
+enum {
+    HEAP_BASE_ID_SYSTEM,
+    HEAP_BASE_ID_SAVE,
+    HEAP_BASE_ID_DEBUG,
+    HEAP_BASE_ID_APPLICATION,
+    HEAP_BASE_ID_COUNT,
+};
```

Comment:
Should these also have a defining label?

Additionally, some conventions:
1. Always explicitly define the initial value.
2. Is `COUNT` meant to be the exclusive upper bound when looping over these values? e.g.:

```c
for (int i = 0; i < HEAP_BASE_ID_COUNT; i++) { /* stuff */ }
```

If so, our convention is that such values should be labeled `MAX`, as it represents a bound and not a total count of values.